homeassistant/components/rtsptowebrtc/__init__.py

```diff
@@ -0,0 +1,103 @@
+"""RTSPtoWebRTC integration with an external RTSPToWebRTC Server.
+
+WebRTC uses a direct communication from the client (e.g. a web browser) to a
+camera device. Home Assistant acts as the signal path for initial set up,
+passing through the client offer and returning a camera answer, then the client
+and camera communicate directly.
+
+However, not all cameras natively support WebRTC. This integration is a shim
+for camera devices that support RTSP streams only, relying on an external
+server RTSPToWebRTC that is a proxy. Home Assistant does not participate in
+the offer/answer SDP protocol, other than as a signal path pass through.
+
+Other integrations may use this integration with these steps:
+- Check if this integration is loaded
+- Call is_suported_stream_source for compatibility
+- Call async_offer_for_stream_source to get back an answer for a client offer
+"""
+
+from __future__ import annotations
+
+import logging
+
+import async_timeout
+from rtsp_to_webrtc.client import Client
+from rtsp_to_webrtc.exceptions import ClientError, ResponseError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady, HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "rtsptowebrtc"
+DATA_SERVER_URL = "server_url"
+DATA_CLIENT = "client"
+DATA_UNAVAILABLE = "unavailable"
+TIMEOUT = 10
+RTSP_PREFIXES = {"rtsp://", "rtsps://"}
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up RTSPtoWebRTC from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    if DATA_SERVER_URL not in entry.data:
+        _LOGGER.error("Invalid ConfigEntry for webrtc, missing '%s'", DATA_SERVER_URL)
+        return False
+
+    client = Client(async_get_clientsession(hass), entry.data[DATA_SERVER_URL])
+    try:
+        async with async_timeout.timeout(TIMEOUT):
+            await client.heartbeat()
+    except ResponseError as err:
+        if DATA_UNAVAILABLE not in hass.data[DOMAIN]:
+            _LOGGER.error("RTSPtoWebRTC server returned failure: %s", err)
+            hass.data[DOMAIN][DATA_UNAVAILABLE] = True
```

Comment:
Can we drop this?

```suggestion
```

`ConfigEntryNotReady` will already take care of the logging.

```
2021-11-24 13:49:44 ERROR (MainThread) [homeassistant.components.rtsptowebrtc] RTSPtoWebRTC server communication failure: RTSPtoWebRTC server communication failure: Cannot connect to host 127.0.0.1:8083 ssl:default [Connect call failed ('127.0.0.1', 8083)]
2021-11-24 13:49:44 WARNING (MainThread) [homeassistant.config_entries] Config entry 'http://127.0.0.1:8083' for rtsptowebrtc integration not ready yet: RTSPtoWebRTC server communication failure: Cannot connect to host 127.0.0.1:8083 ssl:default [Connect call failed ('127.0.0.1', 8083)]; Retrying in background
```

Reply:
Great! My impression is this is a new behavior, so that is great. I will remove this logic from `nest`.